dango/dex/src/core/liquidity_pool.rs

```diff
@@ -263,3 +407,192 @@ fn abs_diff(a: Uint128, b: Uint128) -> Uint128 {
+            (Udec128::new_percent(107), Uint128::from(88168)),
+            (Udec128::new_percent(108), Uint128::from(86535)),
+            (Udec128::new_percent(109), Uint128::from(84947)),
+            (Udec128::new_percent(110), Uint128::from(83403)),
+        ],
+        1 ; "xyk pool balance 1:1 no fee")]
+    #[test_case(
+        CurveInvariant::Xyk {
+            order_depth: 10,
+            order_spacing: Udec128::new_percent(1),
+        },
+        Udec128::new_percent(1),
+        coins! {
+            eth::DENOM.clone() => 10000000,
+            usdc::DENOM.clone() => 10000000,
+        },
+        vec![
+            (Udec128::new_percent(98), Uint128::from(204081)),
+            (Udec128::new_percent(97), Uint128::from(105196)),
+            (Udec128::new_percent(96), Uint128::from(107388)),
+            (Udec128::new_percent(95), Uint128::from(109649)),
+            (Udec128::new_percent(94), Uint128::from(111982)),
+            (Udec128::new_percent(93), Uint128::from(114390)),
+            (Udec128::new_percent(92), Uint128::from(116877)),
+            (Udec128::new_percent(91), Uint128::from(119445)),
+            (Udec128::new_percent(90), Uint128::from(122100)),
+            (Udec128::new_percent(89), Uint128::from(124843)),
+        ],
+        vec![
+            (Udec128::new_percent(102), Uint128::from(196078)),
+            (Udec128::new_percent(103), Uint128::from(95184)),
+            (Udec128::new_percent(104), Uint128::from(93353)),
+            (Udec128::new_percent(105), Uint128::from(91575)),
+            (Udec128::new_percent(106), Uint128::from(89847)),
+            (Udec128::new_percent(107), Uint128::from(88168)),
+            (Udec128::new_percent(108), Uint128::from(86535)),
+            (Udec128::new_percent(109), Uint128::from(84947)),
+            (Udec128::new_percent(110), Uint128::from(83403)),
+            (Udec128::new_percent(111), Uint128::from(81900)),
+        ],
+        1 ; "xyk pool balance 1:1 one percent fee")]
+    fn curve_on_orderbook(
+        curve_invariant: CurveInvariant,
+        swap_fee_rate: Udec128,
+        pool_liquidity: Coins,
+        expected_bids: Vec<(Udec128, Uint128)>,
+        expected_asks: Vec<(Udec128, Uint128)>,
+        order_size_tolerance: u128,
+    ) {
+        let pair = PairParams {
+            curve_invariant,
+            swap_fee_rate: Bounded::new_unchecked(swap_fee_rate),
+            lp_denom: Denom::new_unchecked(vec!["lp".to_string()]),
+        };
+
+        let reserves = pool_liquidity.try_into().unwrap();
+        let (bids, asks) = pair
+            .reflect_curve(eth::DENOM.clone(), usdc::DENOM.clone(), &reserves)
+            .unwrap();
+
+        for (bid, expected_bid) in bids.zip(expected_bids.iter()) {
+            let bid = bid.unwrap();
+            assert_eq!(bid.0, expected_bid.0);
+            assert!(
+                bid.1.into_inner().abs_diff(expected_bid.1.into_inner()) <= order_size_tolerance
+            );
+        }
```

Comment:
Generally I don't like to do `bid.0`, `bid.1`. It's hard to tell what 0 or 1 means.

It's better if you do:

```rust
let (bid_price, bid_amount) = bid;
```

And then use `bid_price`, `bid_amount` instead of `bid.0`, `bid.1`.